### PR TITLE
audit: catch empty installations

### PR DESCRIPTION
**Pull request by xu-cheng**
Create a new PR as the old one is too noise.

Ping @DomT4.

Closes #36642

### CODE REVIEW

File: Library/Homebrew/cmd/audit.rb [hunk 1]
```diff
@@ -668,6 +668,22 @@ def audit_line(line, lineno)
     end
   end
 
+  def audit_prefix_has_contents
+    return unless formula.prefix.directory?
+
+    Pathname.glob("#{formula.prefix}/**/*").each do |file|
```
Comment (jacknagel) [inline Library/Homebrew/cmd/audit.rb]: `glob` can take a block directly (which is both convenient and avoids loading the whole result into memory at once)

Comment (jacknagel) [inline Library/Homebrew/cmd/audit.rb]: ... which should help the pathological case where there are hundreds of formula installed.

Comment (xu-cheng) [inline Library/Homebrew/cmd/audit.rb]: Done.

File: Library/Homebrew/cmd/audit.rb [hunk 2]
```diff
@@ -668,6 +668,22 @@ def audit_line(line, lineno)
     end
   end
 
+  def audit_prefix_has_contents
+    return unless formula.prefix.directory?
+
+    Pathname.glob("#{formula.prefix}/**/*") do |file|
+      next if file.directory?
+      basename = file.basename.to_s
+      return unless Metafiles.copy?(basename) || %w[.DS_Store INSTALL_RECEIPT.json].include?(basename)
```
Comment (mikemcquaid) [inline Library/Homebrew/cmd/audit.rb]: Can we split this into two `returns`; I find `unless` and `||` combined super hard to parse. Thanks!

**Comment by DomT4** [comment]
Thanks all - Happy to let everyone else take it forwards; y'all took my awful implementation and made it much less hideous, so I'm content.

My testing stick has been the prior version of ` osh `, replacing the one in ` HOMEBREW_PREFIX/Library/Formula ` with the one [here](https://raw.githubusercontent.com/Homebrew/homebrew/547f26eab3a97d2932b6724e6317db88af7b447a/Library/Formula/osh.rb) and then doing ` brew install osh ` and ` brew audit osh `. If the audit change is successful, we should get a big red warning about it being empty.

File: Library/Homebrew/cmd/audit.rb (hunk 2 above)
Comment (xu-cheng) [inline Library/Homebrew/cmd/audit.rb]: No it will change the logic. That's the reason @DomT4's test is failing. 

However if you want. I can change it into:
```ruby
next if ...
next if ...
return
```

Comment (mikemcquaid) [inline Library/Homebrew/cmd/audit.rb]: Sorry, yes, that works too, thanks.

Comment (xu-cheng) [inline Library/Homebrew/cmd/audit.rb]: Done.

**Comment by mikemcquaid** [comment]
Nice work here @xu-cheng, @DomT4!
